src/services/auth/auth.service.ts

```diff
@@ -0,0 +1,110 @@
+import { User, UserApi } from '@/stores/user-manager.store.model'
+
+const LOGIN_URL = import.meta.env.VITE_LOGIN_URL
+const LOGOUT_URL = import.meta.env.VITE_LOGOUT_URL
+const USERINFO_URL = import.meta.env.VITE_USERINFO_URL
+
+/**
+ * Calls "/login" url to authenticate user
+ * @param userName The user's name, mandatory
+ * @param userPassword  The user's password, mandatory
+ * @param isApp If the app is mobile app mode, false by default
+ * @returns The api returns user's info is succeeded
+ */
+export async function authenticate(
+  userName: string,
+  userPassword: string,
+  isApp = false
+) {
+  const payload = new URLSearchParams({
+    login: userName,
+    password: userPassword,
+    app: isApp ? 'true' : 'false',
+  })
+  const response = await fetch(LOGIN_URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: payload.toString(),
+  })
+
+  if (!response.ok) {
+    throw new Error('Error while trying to authenticate user')
+  }
+
+  const data = <UserApi>await response.json()
+
+  return mapUserApiToUser(data)
+}
+
+/**
+ * Calls "/logout" url to log out the user
+ * @returns The api returns true if succeeded
+ */
+export async function logout() {
+  const response = await fetch(LOGOUT_URL)
+
+  if (!response.ok) {
+    throw new Error('Error while trying to logout user')
+  }
+
+  const data = await response.text()
+
+  return data
+}
+
+/**
+ * Calls "/getuserinfo" url to get user's info, the user needs to be authenticated first
+ * @returns The api returns user's info is succeeded
+ */
+export async function getUserInfo() {
+  const payload = new URLSearchParams({})
+  const response = await fetch(USERINFO_URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: payload.toString(),
+  })
+
+  if (!response.ok) {
+    throw new Error('Error while trying to get user info')
+  }
+
+  const data = await response.json()
```

Comment:
Actually when the api responds with error 401(invalid user), response.ok is false and this doesn't throw an error and silently fails. I must throw an error in this case because I did a catch in `auth-form` component, see: https://github.com/Geoportail-Luxembourg/luxembourg-geoportail/blob/71ba27766daba80dc99e7ef68360bf43383cb310/src/components/auth/auth-form.vue#L55

Reply:
Oops you are right, I was focused on l. 32 in the `authenticate()` function... so I didn't get your point. I have updated the code as you have suggested.